language/move-core/types/src/language_storage.rs

```diff
@@ -219,4 +297,47 @@ mod tests {
         let c: TypeTag = serde_json::from_str(&b).unwrap();
         assert!(a.eq(&c), "Typetag serde error")
     }
+
+    #[test]
+    fn test_nested_type_tag_struct_serde() {
+        let mut type_tags = vec![make_type_tag_struct(TypeTag::U8)];
+
+        let limit = super::MAX_TYPE_TAG_NESTING - 1;
+        while type_tags.len() < limit.into() {
+            type_tags.push(make_type_tag_struct(type_tags.last().unwrap().clone()));
+        }
+
+        bcs::to_bytes(type_tags.last().unwrap()).unwrap();
+
+        let nest_limit = make_type_tag_struct(type_tags.last().unwrap().clone());
+        bcs::to_bytes(&nest_limit).unwrap_err();
+    }
+
+    #[test]
+    fn test_nested_type_tag_vector_serde() {
+        let mut type_tags = vec![make_type_tag_struct(TypeTag::U8)];
+
+        let limit = super::MAX_TYPE_TAG_NESTING - 1;
+        while type_tags.len() < limit.into() {
+            type_tags.push(make_type_tag_vector(type_tags.last().unwrap().clone()));
+        }
+
+        bcs::to_bytes(type_tags.last().unwrap()).unwrap();
+
+        let nest_limit = make_type_tag_vector(type_tags.last().unwrap().clone());
+        bcs::to_bytes(&nest_limit).unwrap_err();
+    }
+
+    fn make_type_tag_vector(type_param: TypeTag) -> TypeTag {
+        TypeTag::Vector(Box::new(type_param))
+    }
+
+    fn make_type_tag_struct(type_param: TypeTag) -> TypeTag {
+        TypeTag::Struct(StructTag {
+            address: AccountAddress::ONE,
+            module: Identifier::from_utf8(("a".as_bytes()).to_vec()).unwrap(),
```

Comment:
just for future reference and I can only imagine your "care factor" here, but if you want to save your beautiful fingers a bit more you can do `Identifier::new("a").unwrap()`.
I know, I know....

Reply:
copypasta :)